src/Vault.sol

```diff
@@ -310,22 +310,35 @@ contract Vault is IERC4626, ERC20Permit, ILiquidationSource, IClaimable, Ownable
 
   /* ============ Modifiers ============ */
 
-  /// @notice Require reverting if the vault is under-collateralized.
+  /// @notice Modifier reverting if the Vault is under-collateralized.
   modifier onlyVaultCollateralized() {
-    if (!_isVaultCollateralized(_totalSupply(), _totalAssets())) revert VaultUndercollateralized();
+    _onlyVaultCollateralized(_totalSupply(), _totalAssets());
     _;
   }
 
   /**
-   * @notice Requires the caller to be the claimer
+   * @notice Reverts if the Vault is under-collateralized.
+   * @param _depositedAssets Assets deposited into the YieldVault
+   * @param _withdrawableAssets Assets withdrawable from the YieldVault
+   */
+  function _onlyVaultCollateralized(
+    uint256 _depositedAssets,
+    uint256 _withdrawableAssets
+  ) internal view {
+    if (!_isVaultCollateralized(_depositedAssets, _withdrawableAssets))
+      revert VaultUndercollateralized();
+  }
```

Comment:
Can be pure instead of view

Reply:
Fixed in this commit: https://github.com/GenerationSoftware/pt-v5-vault/pull/51/commits/029a830aec2a6519a7c7375dec0ccde99d0e9ef5